scripts/linters/docstrings_checker.py

```
@@ -101,71 +103,84 @@ def returns_something(return_node: astroid.Return) -> bool:
+
+            if handler and handler.type:
+                inferred_excs = astroid.unpack_infer(handler.type)
+                excs = [
+                    exc.name
+                    for exc in inferred_excs
+                    if exc is not astroid.Uninferable
+                ]
+
         return set(
             exc for exc in excs if not utils.node_ignores_exception(node, exc)
         )
```

Ruff (0.15.21)

[warning] Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)